src/main/java/uk/gov/verifiablelog/merkletree/MemoizationStore.java

```diff
@@ -1,7 +1,24 @@
 package uk.gov.verifiablelog.merkletree;
 
+/**
+ * Defines the contract through which a consumer of a {@link MemoizationStore} can query and update a set of known
+ * Merkle Tree root hashes for intermediate subtrees of a particular size and start index.
+ */
 public interface MemoizationStore {
-    public void put(Integer start, Integer size, byte[] value);
-    public byte[] get(Integer start, Integer size);
+    /**
+     * Adds the root hash of a subtree to the set of known intermediate Merkle Tree root hashes.
```

Comment:
I think it should be clear that `put` does not need to always add the root hash of a subtree, and that `get` can always return `null` if it feels like it